server/src/infra/repositories/asset.repository.ts

```diff
@@ -511,19 +515,35 @@ export class AssetRepository implements IAssetRepository {
           longitude: true,
         },
       },
-      where: {
-        ownerId: In([...ownerIds]),
-        isVisible: true,
-        isArchived,
-        exifInfo: {
-          latitude: Not(IsNull()),
-          longitude: Not(IsNull()),
+      where: [
+        {
+          ownerId: In([...ownerIds]),
+          isVisible: true,
+          isArchived,
+          exifInfo: {
+            latitude: Not(IsNull()),
+            longitude: Not(IsNull()),
+          },
+          isFavorite,
+          fileCreatedAt: OptionalBetween(fileCreatedAfter, fileCreatedBefore),
         },
-        isFavorite,
-        fileCreatedAt: OptionalBetween(fileCreatedAfter, fileCreatedBefore),
-      },
+        {
+          albums: {
+            id: In([...albumIds]),
+          },
+          isVisible: true,
+          isArchived,
+          exifInfo: {
+            latitude: Not(IsNull()),
+            longitude: Not(IsNull()),
+          },
+          isFavorite,
+          fileCreatedAt: OptionalBetween(fileCreatedAfter, fileCreatedBefore),
```

Comment:
This section is repeated twice. Can you move it up to a variable?

Reply:
Good idea, done.